bytes/bytes/api/router.py

```diff
@@ -148,42 +148,59 @@ def get_normalizer_meta(
 
 @router.post("/raw", tags=[RAW_TAG])
 def create_raw(
-    request: Request,
     boefje_meta_id: UUID,
-    mime_types: list[str] | None = Query(None),
+    boefje_output: BoefjeOutput,
     meta_repository: MetaDataRepository = Depends(create_meta_data_repository),
     event_manager: EventManager = Depends(create_event_manager),
-) -> RawResponse:
-    parsed_mime_types = [] if mime_types is None else [MimeType(value=mime_type) for mime_type in mime_types]
-
-    try:
-        meta = meta_repository.get_boefje_meta_by_id(boefje_meta_id)
-
-        if meta_repository.has_raw(meta, parsed_mime_types):
-            return RawResponse(status="success", message="Raw data already present")
-
-        # FastAPI/starlette only has async versions of the Request methods, but
-        # all our code is sync, so we wrap it in async_to_sync.
-        data = async_to_sync(request.body)()
-
-        raw_data = RawData(value=data, boefje_meta=meta, mime_types=parsed_mime_types)
-        with meta_repository:
-            raw_id = meta_repository.save_raw(raw_data)
-
-        event = RawFileReceived(
-            organization=meta.organization,
-            raw_data=RawDataMeta(
-                id=raw_id,
-                boefje_meta=raw_data.boefje_meta,
-                mime_types=raw_data.mime_types,
-            ),
-        )
-        event_manager.publish(event)
-    except Exception as error:
-        logger.exception("Error saving raw data")
-        raise HTTPException(status_code=500, detail="Could not save raw data") from error
-
-    return RawResponse(status="success", message="Raw data saved", id=raw_id)
+) -> JSONResponse:
```

Comment:
I suggest using a more specific type for this, as it describes the function better. FastAPI takes care of the rest, producing a correct OpenAPI specification and wrapping it in the right response object

```suggestion
) -> dict[str, str]:
```